### PR TITLE
PFM- 1709 : Individual Enrollment and DeEnrollment from register

### PR DESCRIPTION
**Pull request by harshal-egov**

### CODE REVIEW

File: backend/attendance-service/src/main/java/org/egov/config/AttendanceServiceConfiguration.java [hunk 1]
```diff
@@ -35,26 +35,22 @@ public class AttendanceServiceConfiguration {
     @Value("${egov.mdms.search.endpoint}")
     private String mdmsEndPoint;
     //Topic
-    @Value("${attendance-register.kafka.create.topic}")
+    @Value("${attendance.service.register.kafka.create.topic}")
```
Comment (subhashini-egov) [inline backend/attendance-service/src/main/java/org/egov/config/AttendanceServiceConfiguration.java:38]: We can remove the "service" bit everywhere in the config.

**Review by subhashini-egov — APPROVED** [review]
We will need to schedule an additional review once the attendance module is done as a whole. For now, approved.

File: backend/attendance-service/src/main/java/org/egov/enrichment/AttendeeEnrichmentService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class AttendeeEnrichmentService {
+
+    @Autowired
+    private AttendanceServiceUtil attendanceServiceUtil;
+
+    public void enrichCreateAttendee(AttendeeCreateRequest attendeeCreateRequest, List<IndividualEntry> attendeeListFromDB) {
```
[the remark this answers is not in the document]
Comment (harshal-egov) [reply backend/attendance-service/src/main/java/org/egov/enrichment/AttendeeEnrichmentService.java:22]: Method name Changed.

File: backend/attendance-service/src/main/java/org/egov/enrichment/AttendeeEnrichmentService.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        RequestInfo requestInfo = attendeeCreateRequest.getRequestInfo();
+        List<IndividualEntry> attendeeListFromRequest = attendeeCreateRequest.getAttendees();
+
+        for (IndividualEntry attendeeFromRequest : attendeeListFromRequest) {
```
[the remark this answers is not in the document]
Comment (harshal-egov) [reply backend/attendance-service/src/main/java/org/egov/enrichment/AttendeeEnrichmentService.java:26]: attendeeFromRequest variable renamed as attendee

File: backend/attendance-service/src/main/java/org/egov/enrichment/AttendeeEnrichmentService.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            attendeeFromRequest.setDenrollmentDate(null);
+            if (attendeeFromRequest.getEnrollmentDate() == null) {
+                BigDecimal enrollmentDate = new BigDecimal(System.currentTimeMillis());
+                attendeeFromRequest.setEnrollmentDate(enrollmentDate);
+            }
+
+        }
+    }
+
+    public void enrichDeleteAttendee(AttendeeDeleteRequest attendeeDeleteRequest, List<IndividualEntry> attendeesFromDB) {
```
[the remark this answers is not in the document]
Comment (harshal-egov) [reply backend/attendance-service/src/main/java/org/egov/enrichment/AttendeeEnrichmentService.java:39]: Method name changed to enrichAttendeeOnDelete. In this method we are updating the de-enrollment date (signifies soft delete) and auditdetails (modified by, modified time)

File: backend/attendance-service/src/main/java/org/egov/models/AttendeeSearchCriteria.java [hunk 5]
```diff
@@ -12,20 +13,17 @@
 @Builder
 public class AttendeeSearchCriteria {
 
-    @JsonProperty("id")
-    private List<String> ids;
-
     @JsonProperty("individualId")
```
[the remark this answers is not in the document]
Comment (harshal-egov) [reply backend/attendance-service/src/main/java/org/egov/models/AttendeeSearchCriteria.java]: Changes done.

**Pull request merged**